src/fhirtypes/ElementDefinition.ts

```diff
@@ -1408,16 +1408,42 @@ export class ElementDefinition {
 
   /**
    * If the element has a single type, graft the type's elements into this StructureDefinition as child elements.
+   * If the element is sliced, unfold from the sliced element on the StructureDefinition
+   * If the elemnet is a content reference, unfold from the referenced element on the StructureDefintion
    * @param {Fishable} fisher - A fishable implementation for finding definitions and metadata
    * @returns {ElementDefinition[]} the unfolded elements or an empty array if the type is multi-value or type can't
    *   be resolved.
    */
   unfold(fisher: Fishable): ElementDefinition[] {
     if (
-      this.type.length === 1 &&
-      (this.type[0].profile == null || this.type[0].profile.length <= 1)
+      (this.type?.length === 1 &&
+        (this.type?.[0].profile == null || this.type?.[0].profile.length <= 1)) ||
+      this.contentReference
```

Comment:
No offense, but this `if` statement is kind of ugly.  I'm trying to think of how it can be simplified or made neater.

If the logic made it through `this.type?.length === 1`, then we know `this.type` exists, so we don't need to do `this.type?` in `and`ed phrases after that.  I also think that line 1420 can be simplified.  I *think* this might be equivalent to the above:

```ts
(this.type?.length === 1 && !(this.type[0].profile?.length > 1)) ||
this.contentReference
```

Is that better or is the inversion more confusing?

Reply:
Am I missing something, or is there a reason not to just do:

Reply:
Isn't this the same:
```
(this.type?.length === 1 && this.type[0].profile?.length <= 1) ||
this.contentReference
```
Were you just adding the inversion because it seems more clear?

Reply:
That's not the same because in that one, if `profile` is null then it evaluates to `false`, whereas we want a null profile to evaluate to `true`.  If you wanted to do it that way, we'd have to do something more like:
```ts
(this.type?.length === 1 && (this.type[0].profile ?? []).length <= 1) ||
this.contentReference
```
I'd be fine with that too if you think it is more clear.

Reply:
Ah wait nevermind, they are different if `profile` is null

Reply:
I actually think that one I just did is more clear than my original suggestion.  So it's got my vote for whatever that is worth. ;-)

Reply:
Alright I'm sold, I'll do that last one. And I'll add an instance test